leases/schema/mutations.py

```diff
@@ -189,6 +189,71 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteBerthLeaseMutation()
 
 
+class SwitchBerthMutation(graphene.ClientIDMutation):
+    class Input:
+        old_lease_id = graphene.ID(required=True)
+        new_berth_id = graphene.ID(required=True)
+        switch_date = graphene.Date(
+            description="The date which will mark the end of the old lease and the start of the new lease."
+            "If none is provided, it will default to the day when the mutation is called"
+        )
+
+    old_berth_lease = graphene.Field(BerthLeaseNode)
+    new_berth_lease = graphene.Field(BerthLeaseNode)
+
+    @classmethod
+    @add_permission_required(BerthLease)
+    @change_permission_required(BerthLease)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, old_lease_id, new_berth_id, **input):
+        switch_date = input.get("switch_date") or calculate_berth_lease_start_date()
+
+        old_lease = get_node_from_global_id(
+            info, old_lease_id, only_type=BerthLeaseNode, nullable=False,
+        )
+        new_berth = get_node_from_global_id(
+            info, new_berth_id, only_type=BerthNode, nullable=False,
+        )
+
+        end_date = old_lease.end_date
+        try:
+            contract = old_lease.contract
+        except ObjectDoesNotExist as e:
+            # Flatten all the error messages on a single list
+            raise VenepaikkaGraphQLError(str(e))
+
+        old_lease = terminate_lease(
+            info, old_lease_id, BerthLeaseNode, end_date=switch_date,
+        )
+        comment = _("Lease terminated due to berth switch")
+        if len(old_lease.comment) > 0:
+            old_lease.comment += f"\n{comment}"
+        else:
+            old_lease.comment = comment
+        old_lease.save(update_fields=["comment"])
+
+        try:
+            new_lease = BerthLease.objects.create(
+                berth=new_berth,
+                boat=old_lease.boat,
+                customer=old_lease.customer,
+                renew_automatically=old_lease.renew_automatically,
+                status=LeaseStatus.PAID,
+                start_date=switch_date,
+                end_date=end_date,
+                comment=_("Lease created from a berth switch"),
```

Comment:
Would be nice if comment contains the harbor name, pier id and number of the previous berth